relay-filter/src/config.rs

```diff
@@ -328,15 +482,209 @@ impl FiltersConfig {
     }
 }
 
+/// Configuration of generic filters combined from project and global config.
+///
+/// See [`CombinedFiltersConfigIter`] for details on how to iterate easily
+/// through the applicable filters.
+#[derive(Copy, Clone, Debug)]
+pub(crate) struct CombinedFiltersConfig<'a> {
```

Comment:
The module structure is not great, which required the `pub(crate)` here. Refactoring to a proper structure is out of scope of this PR though, I may approach it in the future.

Reply:
why not just `pub`?

Reply:
It's not needed. `CombinedFiltersConfig` is only used internally in `relay-filter`, so exposing it seems unnecessary. I'm ok making it `pub` if you have a preference for it.

Reply:
I'd prefer to keep this scoped to the module, this has quite some heavy logic attached to it and shouldn't be used outside of the module.

Reply:
Then I would keep it private all together.

Reply:
It's used in `generic.rs`, so it's either `pub(crate)` or `pub`. I'm going to leave it as `pub(crate)` unless there's a strong opinion on a different suggestion. We can revisit the module structure in the future.

Reply:
Can you just move it to the generic module? Kind of fits there.

Reply:
Actually yes, moved it in [b57c122](https://github.com/getsentry/relay/pull/3161/commits/b57c12245eaaafd914ab1719e97e42dfd65356b8).